api/v1alpha1/acl_types.go

```diff
@@ -0,0 +1,24 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package v1alpha1
+
+// +kubebuilder:validation:XValidation:rule="(has(self.allow) || has(self.deny))",message="one of allow or deny must be specified"
+//
+// ACL defines the IP deny/allow configuration.
+type ACL struct {
```

Comment:
this will need to be generic for multiple attributes
```
acl/authorization:
  type: ClientCIDR
  clientCIDR:
    allow:
    - 1.2.3.0/24    
```

or 

```
acl/authorization:
- type: ClientCIDR
  clientCIDR:
  - 1.2.3.0/24
  action: ALLOW

Reply:
@arkodg did you read https://github.com/envoyproxy/gateway/pull/2652#issuecomment-1952652492 any comments to that? I do understand if all others do want everything gathered to single api. But then that also means that the http filter is going to be after all auth modules all called. That will be potential weakness that comes to security.

Reply:
based on the snippet in https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/rbac_filter#role-based-access-control-rbac-filter, I prefer replacing `attributes` with `clientSelectors`

Reply:
@arkodg but in that case I would like to have acl which do 2 things: first it will only allow internal ip addresses and secondly I want to give access to only sub x. In that case I do not have any other option than put all acl rules after jwt and other filters? Or can there be a way to split the configuration to two different filters which is executed in different places?

Reply:
This is a good point, don't have a good answer for this right now, but this becomes another goal of #2571.

Reply:
thinking something like 

```
authorization:
- order: 1
  rules:
  - clientSelectors:
      clientCIDR:
      - 1.2.3.0/24
    action: ALLOW
```

Reply:
thats a good start, we'll have to deal with this in a holistic way for all features, so prefer if we deferred the API for filter ordering here, else this PR will be not make much progress

Reply:
if this split is going to be possible later, I can skip that for now. But to get things forward, I am thinking how the configuration could look like in case of normal rbac policies.

Lets imagine 3 use-cases:
1) I want allow 1.2.3.0/24
2) I want that only JWT role admin is allowed to access all paths in service
3) I want that JWT role user is allowed to access all paths in service instead /admin/*

```
authorization:
  rules:
  - clientSelectors:
      clientCIDR:
      - 1.2.3.0/24
    permissions:
    - any: true
    action: ALLOW
  - clientSelectors:
      jwt:
        path:
        - payload
        - role
        value:
          stringMatch:
            exact: admin
    permissions:
    - any: true
    action: ALLOW
  - clientSelectors:
      any: true
    permissions:
    - not:
        header:
          name: ":path"
          regexMatch: "/admin(/.*)?"
    action: ALLOW
```

how does that look like?

Reply:
two selectors: the policy does not need that much rows and is less verbose

```
authorization:
  rules:
  - clientSelectors: (can we allow multiple selectors under single rule item?)
      clientCIDR:
      - 1.2.3.0/24
      jwt:
        path:
        - payload
        - role
        value:
          stringMatch:
            exact: admin
    permissions:
    - any: true
    action: ALLOW
  - clientSelectors:
      any: true
    permissions:
    - not:
        header:
          name: ":path"
          regexMatch: "/admin(/.*)?"
    action: ALLOW
```

Reply:
yah, we should allow multiple selectors under a rule`, it mimics the policies field in https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/rbac/v3/rbac.proto#envoy-v3-api-msg-config-rbac-v3-rbac, all selectors should be `AND`d together